_rules/summary-non-empty-accessible-name-2t702h.md

```diff
@@ -0,0 +1,195 @@
+
+If the `summary` element is not included in the accessibility tree, but is still included in sequential focus navigation, this can result in accessibility issues not tested by this rule. This is covered under [Element with aria-hidden has no content in sequential focus navigation][6cfa84].
+
+Note that some user agents expose the `summary` element with a `button` role. This deviates from the implicit ARIA semantics described in [ARIA in HTML](https://www.w3.org/TR/html-aria/#el-summary). Because some browsers do not give `summary` elements a button role, these elements need to be tested separately from the [Button has non-empty accessible name](https://www.w3.org/WAI/standards-guidelines/act/rules/97a4e1/) ACT rule.
+
+### Bibliography
+
+- [Understanding Success Criterion 4.1.2: Name, Role, Value](https://www.w3.org/WAI/WCAG22/Understanding/name-role-value)
+- [ARIA14: Using aria-label to provide an invisible label where a visible label cannot be used](https://www.w3.org/WAI/WCAG22/Techniques/aria/ARIA14)
+- [ARIA16: Using aria-labelledby to provide a name for user interface controls](https://www.w3.org/WAI/WCAG22/Techniques/aria/ARIA16)
+- [Summary element entry for ARIA in HTML](https://www.w3.org/TR/html-aria/#el-summary)
+
+## Test Cases
+
+### Passed
+
+#### Passed Example 1
+
+This `summary` element has an [accessible name][] because of its text content.
+
+```html
+<details>
+	<summary>Opening times</summary>
+	<p>This is a website. We are available 24/7.</p>
+</details>
+```
+
+#### Passed Example 2
+
+This `summary` element has an [accessible name][] because of its `aria-label` attribute.
+
+```html
+<details>
+	<summary aria-label="Opening times"></summary>
+	<p>This is a website. We are available 24/7.</p>
+</details>
+```
+
+#### Passed Example 3
+
+This `summary` element has an [accessible name][] because of its `aria-labelledby` attribute.
+
+```html
+<span id="opening-times">Opening times</span>
+<details>
+	<summary aria-labelledby="opening-times"></summary>
+	<p>This is a website. We are available 24/7.</p>
+</details>
+```
+
+#### Passed Example 4
+
+This `summary` element has an [accessible name][] because of its text content. It does not need to be the first child element of `details`.
+
+```html
+<details>
+	<p>This is a website. We are available 24/7.</p>
+	<summary>Opening times</summary>
+</details>
+```
+
+#### Passed Example 5
+
+This first `summary` element has an [accessible name][] because of its text content. The second `summary` element is inapplicable because only the first `summary` element will be used as a control for the `details` element.
+
+```html
+<details>
+	<summary>Opening times</summary>
+	<summary></summary>
+	<p>This is a website. We are available 24/7.</p>
+</details>
+```
+
+### Failed
+
+#### Failed Example 1
+
+This `summary` element has no [accessible name][] because it has no content or attribute that can provide it.
```

Comment:
```suggestion
This `summary` element has no [accessible name][], or an accessible name with just the marker pseudo element, because it has no content or attribute that can provide it.
```